src/actions/project-text-search.js

```diff
@@ -34,17 +35,21 @@ export function clearSearchResults() {
   };
 }
 
+export function updateStatus(status: string) {
+  return { type: "UPDATE_STATUS", status };
+}
+
 export function searchSources(query: string) {
   return async ({ dispatch, getState }: ThunkArgs) => {
     await dispatch(clearSearchResults());
     await dispatch(addSearchQuery(query));
+    dispatch(updateStatus(statusEnum.fetching));
```

Comment:
can we set the status to fetching when we set the query or done when we set the results?

Reply:
I thought of that at first but this seemed more explicit than your approach